pkg/controller/metrics/custom_collectors_test.go

```diff
@@ -691,32 +696,104 @@ func TestClusterSyncCollector(t *testing.T) {
 			existing: []runtime.Object{
 				testcs.FullBuilder("test-namespace", "test-name", scheme).Options(FailingSince(time.Now())).Build(),
 			},
-			min:       1 * time.Hour,
-			expected1: []string(nil),
-			expected2: []string(nil),
+			min:            1 * time.Hour,
+			optionalLabels: map[string]string{},
+			expected1:      []string(nil),
+			expected2:      []string(nil),
 		},
 		{
 			name: "clustersync passed threshold",
 			existing: []runtime.Object{
 				testcs.FullBuilder("test-namespace", "test-name", scheme).Options(FailingSince(time.Now())).Build(),
+				cdBuilder("test-namespace").Build(testcd.WithCondition(hivev1.ClusterDeploymentCondition{
+					Type:   hivev1.UnreachableCondition,
+					Status: corev1.ConditionUnknown,
+				})),
+			},
+			min:            0 * time.Hour,
+			optionalLabels: map[string]string{},
+			expected1:      []string{"namespaced_name = test-namespace/test-name unreachable = Unknown"},
+			expected2:      []string(nil),
+		},
+		{
+			name:           "no clustersync",
+			existing:       nil,
+			min:            1 * time.Hour,
+			optionalLabels: map[string]string{},
+			expected1:      []string(nil),
+			expected2:      []string(nil),
+		},
+		{
+			name: "report expected fixed labels",
+			existing: []runtime.Object{
+				testcs.FullBuilder("test-namespace", "test-name", scheme).Options(FailingSince(time.Now())).Build(),
+				cdBuilder("test-namespace").Build(testcd.WithCondition(hivev1.ClusterDeploymentCondition{
+					Type:   hivev1.UnreachableCondition,
+					Status: corev1.ConditionTrue,
+				})),
 			},
-			min:       0 * time.Hour,
-			expected1: []string{"namespaced_name = test-namespace/test-name"},
+			min:            0 * time.Hour,
+			optionalLabels: map[string]string{},
+			expected1:      []string{"namespaced_name = test-namespace/test-name unreachable = True"},
+			expected2:      []string(nil),
+		},
+		{
+			name: "skip reporting when ClusterDeployment not found",
```

Comment:
Let's add one for when the CD exists, but has no `unreachable` condition.

Reply:
Added, along with ones for the clusterdeployment not found.